### DISCUSSION TITLE
希望能添加的实用功能

### DISCUSSION THREAD

**Discussion by yuyong280** [python-visualization/folium#2022] [Ideas]
希望在LayerControl  中添加，双击图层能 将图层中数据  最大化显示， 谢谢。    
目前我尝试很久未实现。  多谢大佬的分享。

**Comment by hansthen** [comment]
Hi yuyong280, would it be possible for you to communicate in English? Most people actively involved in Folium, don't speak much Chinese. Just to confirm what you want; if you select a Layer in LayerControl you want to limit the viewport to the data in that Layer? Or do you want the viewport limited to all the visible Layers?

**Reply by yuyong280** [reply]
Hello, nice to see your reply. I tried a translation software this time.
I hope to double-click on one of the layers in LayerControl so that the map can maximize the display of the data in that layer. 
It's a bit similar to folium.FitOverlays(fly=True, padding=20).add_to(map).But the way the selection is displayed is different.
Thank you.

**Reply by yuyong280** [reply]
It doesn't change whether the layer is displayed or not, it just changes the field of view to maximize the display of a certain layer of data.

**Comment by hansthen** [comment]
You could add an EventHandler to the Layer object. This will require you to write some JavaScript code as a python `JsCode` object. Something like this:

```
handler = JsCode("""
function onDblClick(event) {
     var layer = event.sourceTarget;
     layer._map.fitBounds(layer.getBounds());

}
""")
layer.on(dblclick=handler)
```

I don't know how well you know JavaScript. I did not test the code above, but it should come close. The `getBounds` method will only exist for some layer types. If you also want to zoom to markers, then maybe have a look at this stack overflow entry: https://stackoverflow.com/questions/48954616/leaflet-layer-getbounds-not-a-function. For extra cool points, you can use `flyTo`.

**Reply by yuyong280** [reply]
Thank you for your reply. I am an architect . I want to create a project map. After learning about folium.

handler = JsCode("""
function onDblClick(event) {
var target = event.target;
if (target instanceof L.Marker || target instanceof L.FeatureGroup) {
target._map.fitBounds(target.getBounds());
}
}
""")
layer.on(dblclick=handler, bubble=True)

I tried your code and asked AI. Implemented double clicking on the data in the image to zoom in to that data.
But I still can't zoom to that layer by double clicking on the layer in LayerControl().

**Reply by hansthen** [reply]
Do you mean that you want to double click on the layer's label inside of the LayerControl()?

**Reply by hansthen** [reply]
Perhaps you can create a short screen recording of what you want to achieve?

**Reply by yuyong280** [reply]
double click on the layer's label inside of the LayerControl()

**Reply by yuyong280** [reply]
![微信图片_20241118143410](https://github.com/user-attachments/assets/1009c604-6dca-435e-ac55-a7198a34e459)

**Reply by hansthen** [reply]
I understand. I have to find some time to look into this. I think it should be possible to do this, but I am not sure it can be done using only `Folium`. You could probably checkout what events are generated by the upstream `Leaflet.LayerControl` and see if it generates an event for double click.
